Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 54)
+++ CMakeLists.txt	(working copy)
@@ -155,7 +155,7 @@
 if( NOT $ENV{BOOST_LIBRARYDIR} STREQUAL "" )
   set(Boost_LIBRARY_DIRS $ENV{BOOST_LIBRARYDIR})
 endif()
-find_package(Boost COMPONENTS python)
+find_package(Boost COMPONENTS system python)
 
 if( Boost_FOUND )
   message(STATUS "found boost version: ${Boost_VERSION}")
@@ -223,7 +223,7 @@
 
     add_definitions(${Boost_CFLAGS})
     set(CMAKE_REQUIRED_INCLUDES ${PYTHON_INCLUDE_PATH} ${Boost_INCLUDE_DIR} )
-    set(CMAKE_REQUIRED_LIBRARIES ${PYTHON_LIBRARIES} ${Boost_PYTHON_LIBRARY} ${Boost_THREAD_LIBRARY})
+    set(CMAKE_REQUIRED_LIBRARIES ${PYTHON_LIBRARIES} ${Boost_PYTHON_LIBRARY} ${Boost_SYSTEM_LIBRARY})
     set(CMAKE_REQUIRED_FLAGS ${Boost_CFLAGS})
 
     # check if all header files can be included
